crates/evm/evm/src/executors/fuzz/mod.rs

```diff
@@ -214,7 +214,8 @@ impl FuzzedExecutor {
         );
 
         // When the `assume` cheatcode is called it returns a special string
-        if call.result.as_ref() == MAGIC_ASSUME {
+        // NOTE: `ends_with` is used to prevent padded results issues
+        if call.result.ends_with(MAGIC_ASSUME) {
```

Comment:
I know I suggested `ends_with` but I liked your way better (converting to U256):
more efficient than a search like this, but it also avoids a case where (I wouldn't know why) a function reverts with a reason that ends with this

Reply:
Unfortunately making this change adds a lot of complexity, reduces performance and code legibility. I believe this solution might be robust enough for our application.